src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/rc_command_request.h

```diff
@@ -0,0 +1,151 @@
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_RPC_PLUGINS_RC_RPC_PLUGIN_INCLUDE_RC_RPC_PLUGIN_COMMANDS_RC_COMMAND_REQUEST_H
+#define SRC_COMPONENTS_APPLICATION_MANAGER_RPC_PLUGINS_RC_RPC_PLUGIN_INCLUDE_RC_RPC_PLUGIN_COMMANDS_RC_COMMAND_REQUEST_H
+
+#include "rc_rpc_plugin/resource_allocation_manager.h"
+#include "rc_rpc_plugin/rc_app_extension.h"
+#include "application_manager/commands/command_request_impl.h"
+
+namespace rc_rpc_plugin {
+namespace app_mngr = application_manager;
+
+enum TypeAccess { kDisallowed, kAllowed };
+
+namespace commands {
+
+class RCCommandRequest : public app_mngr::commands::CommandRequestImpl {
+ public:
+  /**
+   * @brief RCCommandRequest class constructor
+   *
+   **/
+  RCCommandRequest(
+      rc_rpc_plugin::ResourceAllocationManager& resource_allocation_manager,
+      const app_mngr::commands::MessageSharedPtr& message,
+      app_mngr::ApplicationManager& application_manager);
+
+  virtual ~RCCommandRequest();
+
+  virtual void OnTimeout();
+
+  void Run();
+
+  virtual void on_event(const app_mngr::event_engine::Event& event);
+
+ protected:
+  ResourceAllocationManager& resource_allocation_manager_;
+  bool auto_allowed_;
+
+  /**
+   * @brief AcquireResource try to allocate resource for application
+   * In case if allocation of resource is not required, return ALLOWED by
+   * default.
+   * This method should be overrided in RPCs that requires resource allocation
+   * @return result of resource allocation, in case if allocation os not
+   * required, return ALLOWED
+   */
+  virtual AcquireResult::eType AcquireResource(
+      const app_mngr::commands::MessageSharedPtr& message) {
+    return AcquireResult::ALLOWED;
+  }
+
+  /**
+   * @brief IsResourceFree check resource state
+   * This is default implementation which has to be redefined for RPCs which
+   * need to manage the resources
+   * @param module_type Resource name
+   * @return True if free, otherwise - false
+   */
+  virtual bool IsResourceFree(const std::string& module_type) const {
+    UNUSED(module_type);
+    return true;
+  }
+  /**
+   * @brief SetResourceState changes state of resource
+   * This is default implementation which has to be redefined for RPCs which
+   * need to manage the resources
+   * @param Message containing type of module to extract
+   * @param State to set for resource
+   */
+  virtual void SetResourceState(const std::string& module_type,
+                                const ResourceState::eType) {}
+
+
+  TypeAccess CheckModule(const std::string& module_type,
+                         application_manager::ApplicationSharedPtr app);
+
+  bool auto_allowed() const {
+    return auto_allowed_;
+  }
+
+  void set_auto_allowed(bool value) {
+    auto_allowed_ = value;
+  }
+
+  /**
+   * @brief executes specific logic of children classes
+   */
+  void virtual Execute() = 0;
```

Comment:
@LitvinenkoIra is Execute there is virtual. Shouldn't it override Run Method?

Reply:
@LuxoftAKutsan Execute is virtual because it's overridden in children classes. And Execute is called in Run method